[Dynamo] Enable torch._dynamo.config.suppress_errors by default

yanboliang opened the pull request:
Summary:
We are working toward full model compilation, where when compilation error happens, we just fall back to eager mode rather than error out.
But at the same time, we should fix these issues if they are bugs. We will:
* 1/ log warnings in OSS;
* 2/ log warnings and write them into Scuba in fbcode;

to prevent us from ignoring these issues.

Test Plan: Manual test

Differential Revision: D47506314



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @anijain2305

jansel commented on torch/_dynamo/exc.py:
```diff
@@ -226,10 +226,10 @@ def filter_stack(stack):
 
 
-def format_error_msg(exc, code, record_filename=None, frame=None):
+def format_error_msg(exc, code, record_filename=None, frame=None, verbose=config.verbose):
```
this will read `config.verbose` at program startup, but then ignore any changes to it after that.

ezyang commented:
What's the oncall monitoring plan for the scuba stats?

yanboliang commented:
> What's the oncall monitoring plan for the scuba stats?

1/ I'll add one more responsibility for PT2 internal oncall and update runbook;
2/ I'll regularly check the scuba table, triage and fix these issues.

jansel commented on torch/_dynamo/config.py:
```diff
@@ -98,7 +98,7 @@
 # lose optimization opportunities this way.  Devs, if your benchmark model is failing
 # this way, you should figure out why instead of suppressing it.
-suppress_errors = bool(os.environ.get("TORCHDYNAMO_SUPPRESS_ERRORS", False))
+suppress_errors = bool(os.environ.get("TORCHDYNAMO_SUPPRESS_ERRORS", True))
```
```suggestion
suppress_errors = os.environ.get("TORCHDYNAMO_SUPPRESS_ERRORS", "1")=="1"
```

Prior code would confusingly mean `TORCHDYNAMO_SUPPRESS_ERRORS=0` does nothing.

yanboliang committed:
[Dynamo] Enable torch._dynamo.config.suppress_errors by default (#105307)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/105307

We are working toward full model compilation, where when compilation error happens, we just fall back to eager mode rather than error out.
But at the same time, we should fix these issues if they are bugs. We will:
* 1/ log warnings in OSS;
* 2/ log warnings and write them into Scuba in fbcode;

to prevent us from ignoring these issues.

Test Plan: {F1051533782}

Differential Revision: D47506314

fbshipit-source-id: 3dd990aa15c678cc9e308eea608a4778cc9cad1f

yanboliang commented:
@jansel kindly remind in case you miss this.
